WordPress/openverse
# Fix documentation rate limiting and formatting issues; fail docs build in CI if docs build has warnings

## Fixes



Fixes #1972 by @sarayourfriend
Fixes #1971 by @sarayourfriend 
Fixes #1860 by @sarayourfriend
Fixes #1861 by @sarayourfriend

## Description


- Fails the documentation build in CI if warnings exist during the build
- Fixes all toctree issues by manually writing all toc trees. Combined with the previous item, there's no need to have general glob usage to ensure documents are included. If they're not included, the build will fail in CI.
- Adds a `LINK_ISSUE_GITHUB_TOKEN` environment variable to prevent rate limiting from breaking the issue references during build. Disables issue link resolution locally to speed up local build. Adds a documentation page for the link issues plugin and the rationale for the changes.
- Update release drafter templates to use static links for changelog PR links going forward. This prevents the number of issues that `link_issues` needs to process each time the docs are built from growing over time. One can foresee a circumstance where eventually we have so many changelogs that even with the GitHub token we get rate limited. Besides the rate limiting, each new issue reference requires a new API request to GitHub and slows down the documentation build process. Issue to retroactively apply the template changes to existing changelogs is here: https://github.com/WordPress/openverse/issues/1974
- Fixes all cross-reference issues
- Normalise project planning document titles
- Adds checkmarks for approved reviewers and updates the reviewer list when needed
- Add all documents that were previously incorrectly excluded from the documentation


## Testing Instructions

Check out the docs preview generated for this PR. To test the changes to the `link_issues` plugin, the best place to look are on the changelogs. In the currently published version of the docs, several changelog pages do not have properly resolved issue links (see issue for an example). There should no longer be any issues that do not have links properly resolved.

Run `just documentation/build` locally and confirm there are no more errors to resolve.

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
